### PR TITLE
SOLR-13437: fork noggit code into Solr

### PR DESCRIPTION
**Pull request by noblepaul**
- direct use of noggit is a forbidden API now
- I can't remove the dependency because spatial4j uses it

### CODE REVIEW

File: lucene/tools/forbiddenApis/solr.txt [hunk 1]
```diff
@@ -55,3 +55,8 @@ com.google.common.base.Preconditions#checkNotNull(java.lang.Object,java.lang.Obj
 @defaultMessage Use methods in java.util.Comparator instead
 com.google.common.collect.Ordering
 
+@defaultMessage Use corresponding classes in package org.apache.solr.common.json instead
```
Comment (uschindler) [inline lucene/tools/forbiddenApis/solr.txt]: I'd propose to just add `org.noggit.**`, this disallows whole package.

Comment (noblepaul) [reply lucene/tools/forbiddenApis/solr.txt]: sure. thanks . I didn't know it was possible

File: solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
[the remark this answers is not in the document]
Comment (noblepaul) [reply solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java]: ASL let's you modify content , yeah, we can add a credit in the NOTICE.txt

[the remark this answers is not in the document]
Comment (madrob) [reply solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java]: Modify content? Yes. Remove copyright? No. ALv2 section 4c states:

>You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works; and

That pretty clearly reads to me like the copyright attribution has to stay with the source.

Section 4d goes on to state that we only have to carry forward content in NOTICE if the original source has a NOTICE (which this doesn't), so I think that was a misread on my part earlier.

[the remark this answers is not in the document]
Comment (noblepaul) [reply solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java]: I'm not clear what else needs to be done

[the remark this answers is not in the document]
Comment (madrob) [reply solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java]: The files that had a copyright claim in the header need to continue to have the copyright attribution, namely ` *  Copyright 2006- Yonik Seeley`

[the remark this answers is not in the document]
Comment (noblepaul) [reply solr/solrj/src/java/org/apache/solr/common/json/ObjectBuilder.java]: done

**Pull request merged**